File: frontend/src/domain/usecases/RefreshTokenUseCase.ts
import type { IAuthRepository } from "@repositories/IAuthRepository";
import type { AuthToken } from "@entities/AuthToken";
import { MESSAGES } from "@constants/messages";

export class RefreshTokenUseCase {
  private authRepository: IAuthRepository;

  constructor(authRepository: IAuthRepository) {
    this.authRepository = authRepository;
  }

  async execute(refreshToken: string): Promise<AuthToken> {
    if (!refreshToken) {
      throw new Error(MESSAGES.AUTH.REFRESH_TOKEN_MISSING);
    }

    return this.authRepository.refreshToken(refreshToken);
  }
}

File: backend/src/domain/usecases/refresh_token_usecase.py
from domain.entities import AuthToken
from domain.repositories import IAuthRepository


class RefreshTokenUseCase:
    """Use case for refreshing authentication tokens."""

    def __init__(self, auth_repository: IAuthRepository):
        self.auth_repository = auth_repository

    async def execute(self, refresh_token: str) -> AuthToken:
        """Execute refresh token use case."""
        if not refresh_token:
            raise ValueError("Refresh token is required")

        return await self.auth_repository.refresh_token(refresh_token)
